biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -59,74 +61,79 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 		if (header.size() == 0)
 			return false;
 
-		Parameters optionsHeader = OSGiHeader
-			.parseHeader(analyzer.mergeProperties(Constants.CDIANNOTATIONS_OPTIONS));
-		EnumSet<Options> options = EnumSet.noneOf(Options.class);
-		for (Map.Entry<String, Attrs> entry : optionsHeader.entrySet()) {
-			try {
-				Options.parseOption(entry, options, this);
-			} catch (IllegalArgumentException e) {
-				analyzer.error("Unrecognized %s value %s with attributes %s, expected values are %s",
-					Constants.CDIANNOTATIONS_OPTIONS, entry.getKey(), entry.getValue(), EnumSet.allOf(Options.class));
-			}
-		}
-
-		if (options.isEmpty()) {
-			// set the default mode, if org.osgi.service.cdi.annotations.Bean is
-			// on the classpath set to `annotated_by_bean`, otherwise set to
-			// `none`
-
-			TypeRef typeRef = analyzer.getTypeRefFromFQN("org.osgi.service.cdi.annotations.Bean");
-			if (typeRef != null)
-				options.add(Options.annotated_by_bean);
-			else
-				options.add(Options.none);
-		}
-
-		if (options.contains(Options.none)) {
-			return false;
-		}
-
 		Instructions instructions = new Instructions(header);
 		Collection<Clazz> list = analyzer.getClassspace()
 			.values();
+		TypeRef beanTypeRef = analyzer.getTypeRefFromFQN("org.osgi.service.cdi.annotations.Bean");
 		List<String> names = new ArrayList<>();
-
 		TreeSet<String> provides = new TreeSet<>();
 		TreeSet<String> requires = new TreeSet<>();
 
 		for (Clazz c : list) {
-			for (Instruction instruction : instructions.keySet()) {
+			for (Entry<Instruction, Attrs> entry : instructions.entrySet()) {
+				Instruction instruction = entry.getKey();
+				Attrs attrs = entry.getValue();
 				if (instruction.matches(c.getFQN())) {
 					if (instruction.isNegated()) {
 						break;
 					}
+
+					String discover = attrs.get("discover");
+					EnumSet<Discover> options = EnumSet.noneOf(Discover.class);
+					try {
+						Discover.parse(discover, options, this);
+					} catch (IllegalArgumentException e) {
+						analyzer.error("Unrecognized discover '%s', expected values are %s", discover,
+							EnumSet.allOf(Discover.class));
+					}
+
+					if (options.isEmpty()) {
+						// set the default mode, if
+						// org.osgi.service.cdi.annotations.Bean is
+						// on the classpath set to `annotated_by_bean`,
+						// otherwise set to
+						// `none`
+
+						if (beanTypeRef != null)
```

Comment:
As mentioned above, this variable is always non-null. If you want to know if the referenced type is visible, you will need to call analyzer.findClass. Also, not sure why it should vary by the visibility of a class to the analyzer since Bnd does not need to have access to the class to process CDI annotations.